apps/heft/src/plugins/ApiExtractorPlugin/ApiExtractorRunner.ts

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import { Colors, IColorableSequence, ITerminalProvider, Terminal } from '@rushstack/node-core-library';
+import { ApiExtractor as TApiExtractor } from '@microsoft/rush-stack-compiler-3.7';
+
+import { SubprocessRunnerBase } from '../../utilities/subprocess/SubprocessRunnerBase';
+import { PrefixProxyTerminalProvider } from '../../utilities/PrefixProxyTerminalProvider';
+
+export interface IApiExtractorRunnerConfiguration {
+  configFileLocation: string;
+  apiExtractorPackagePath: string;
+  typescriptPackagePath: string;
+  buildFolder: string;
+  production: boolean;
+}
+
+export class ApiExtractorRunner extends SubprocessRunnerBase<IApiExtractorRunnerConfiguration> {
+  private _terminal: Terminal;
+
+  public static getTerminal(terminalProvider: ITerminalProvider): Terminal {
+    const prefixTerminalProvider: PrefixProxyTerminalProvider = new PrefixProxyTerminalProvider(
+      terminalProvider,
+      '[api-extractor] '
+    );
+
+    return new Terminal(prefixTerminalProvider);
+  }
+
+  public get filename(): string {
+    return __filename;
+  }
+
+  public initializeTerminal(terminalProvider: ITerminalProvider): void {
+    this._terminal = ApiExtractorRunner.getTerminal(terminalProvider);
+  }
+
+  public async invokeAsync(): Promise<void> {
+    const apiExtractor: typeof TApiExtractor = require(this._configuration.apiExtractorPackagePath);
+    const extractorConfig: TApiExtractor.ExtractorConfig = apiExtractor.ExtractorConfig.loadFileAndPrepare(
+      this._configuration.configFileLocation
+    );
+    const extractorOptions: TApiExtractor.IExtractorInvokeOptions = {
+      localBuild: !this._configuration.production,
+      typescriptCompilerFolder: this._configuration.typescriptPackagePath,
+      messageCallback: (message: TApiExtractor.ExtractorMessage) => {
+        switch (message.logLevel) {
+          case apiExtractor.ExtractorLogLevel.Error: {
+            let logMessage: (IColorableSequence | string)[];
+            if (message.sourceFilePath) {
+              const buildFolderRelativeFilePath: string = path.relative(
+                this._configuration.buildFolder,
+                message.sourceFilePath
+              );
+              logMessage = [
+                Colors.red(
+                  `${buildFolderRelativeFilePath}(${message.sourceFileLine},${message.sourceFileColumn})`
+                ),
+                Colors.white(' - '),
+                Colors.gray(`(${message.category})`),
+                ' ',
+                Colors.red(message.text)
+              ];
+            } else {
+              logMessage = [message.text];
+            }
+
+            this._terminal.writeErrorLine(...logMessage);
```

Comment:
The real API Extractor adds `Error: ` and `Warning: ` prefixes. Should we do that here?